### PR TITLE
Fix kubeadm api codegen

### PR DESCRIPTION
**Pull request by jbeda**
Codegen was broken because it looks for a magic file named `doc.go` not
`docs.go`.  Also had to harmonize types to make conversion work well.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/34497)

### CODE REVIEW

**Comment by mikedanese** [comment]
Thanks for the fix!

File: cmd/kubeadm/app/apis/kubeadm/doc.go [hunk 1]
```diff
@@ -15,7 +15,6 @@ limitations under the License.
 */
 
 // +k8s:deepcopy-gen=package,register
-// +k8s:conversion-gen=k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm
```
Comment (mikedanese) [inline cmd/kubeadm/app/apis/kubeadm/doc.go]: why remove this?

**Comment by mikedanese** [comment]
I'm going to fix the error and add kubeadm api group to round trip conversion testing.

**Comment by mikedanese** [comment]
@jbeda can you drop the generated code files from this PR and just remove the `// +k8s:` macros from both doc.go and v1alpha1/doc.go? I don't think we need generated conversions or deep-copies.

**Comment by jbeda** [comment]
Ok. Will do so in a bit.

**Comment by jbeda** [comment]
Superseded by #34555

**Pull request closed without merging**